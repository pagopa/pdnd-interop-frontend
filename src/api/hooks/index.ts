export * from './useDownloadFile'
export * from './useAuthenticatedQuery'
