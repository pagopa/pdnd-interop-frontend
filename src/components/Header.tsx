import React, { useContext } from 'react'
import { useLocation } from 'react-router-dom'
import { Typography } from '@mui/material'
import { Box } from '@mui/system'
import { PartyContext, TokenContext } from '../lib/context'
import { showPlatformTwoColumnsLayout } from '../lib/router-utils'
import { Layout } from './Shared/Layout'
import { StyledButton } from './Shared/StyledButton'
import { StyledLink } from './Shared/StyledLink'
import { PartySelect } from './PartySelect'
import logo from '../assets/pagopa-logo-white.svg'
import { ROUTES } from '../config/routes'
import { URL_FE_LOGIN } from '../lib/constants'

export function Header() {
  const location = useLocation()
  const { party } = useContext(PartyContext)
  const { token } = useContext(TokenContext)

  const { PATH: btnPath, LABEL: btnLabel } = token
    ? ROUTES.LOGOUT
    : { PATH: URL_FE_LOGIN, LABEL: 'Login' }

  return (
    <header>
      <Box sx={{ bgcolor: 'primary.dark' }}>
        <Layout>
          <Box
            sx={{
              display: 'flex',
              justifyContent: 'space-between',
              alignItems: 'center',
              py: 3,
            }}
          >
            <StyledLink to="/" sx={{ display: 'flex' }}>
              <Box component="img" src={logo} alt="Logo PagoPA" sx={{ margin: 'auto' }} />
            </StyledLink>
            <StyledButton variant="contained" to={btnPath}>
              {btnLabel}
            </StyledButton>
          </Box>
        </Layout>
      </Box>

      <Box sx={{ bgcolor: 'primary.main' }}>
        <Layout>
          <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
            <Box sx={{ py: 3 }}>
              <Typography color="common.white" fontWeight={700} fontSize={24}>
                Interoperabilit√†
              </Typography>
              <Typography color="common.white" variant="caption">
                Il catalogo degli e-service delle PA
              </Typography>
            </Box>

            {showPlatformTwoColumnsLayout(location) && party !== null && <PartySelect />}
          </Box>
        </Layout>
      </Box>
    </header>
  )
}
