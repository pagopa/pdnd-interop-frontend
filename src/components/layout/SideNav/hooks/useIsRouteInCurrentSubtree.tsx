import { type RouteKey, getParentRoutes, useCurrentRoute } from '@/router'

export function useIsRouteInCurrentSubtree() {
  const { routeKey: currentRouteKey } = useCurrentRoute()

  console.log('currentRouteKey', currentRouteKey)
  return (routeKey: RouteKey) => {
    return [...getParentRoutes(currentRouteKey), currentRouteKey].includes(routeKey)
  }
}
