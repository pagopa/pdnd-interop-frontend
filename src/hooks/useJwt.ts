import React from 'react'
import { useAuth } from '@/stores'
import type { JwtUser } from '@/types/party.types'
import memoize from 'lodash/memoize'

const parseJwt = memoize((token: string | null) => {
  const jwt = token ? (JSON.parse(atob(token.split('.')[1])) as JwtUser) : undefined
  const currentRoles = jwt ? jwt.organization.roles.map((r) => r.role) : []
  const isAdmin = currentRoles.length === 1 && currentRoles[0] === 'admin'
  const isOperatorAPI = Boolean(currentRoles.includes('api'))
  const isOperatorSecurity = Boolean(currentRoles.includes('security'))

  return { jwt, currentRoles, isAdmin, isOperatorAPI, isOperatorSecurity }
})

export const useJwt = () => {
  const { sessionToken } = useAuth()

  const { jwt, currentRoles, isAdmin, isOperatorAPI, isOperatorSecurity } = parseJwt(sessionToken)

  const hasSessionExpired = React.useCallback(
    () => (jwt ? new Date() > new Date(jwt.exp * 1000) : false),
    [jwt]
  )

  const isCurrentUser = React.useCallback(
    (userId: string) => {
      return jwt && jwt.uid === userId
    },
    [jwt]
  )

  return {
    jwt,
    isCurrentUser,
    hasSessionExpired,
    isAdmin,
    isOperatorAPI,
    isOperatorSecurity,
    currentRoles,
  }
}
