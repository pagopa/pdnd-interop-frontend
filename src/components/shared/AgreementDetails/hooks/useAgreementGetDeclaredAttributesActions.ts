import { isAttributeOwned } from '@/utils/attribute.utils'
import { useAgreementDetailsContext } from '../AgreementDetailsContext'
import { useTranslation } from 'react-i18next'
import { useCurrentRoute } from '@/router'
import { AttributeMutations } from '@/api/attribute'
import { AuthHooks } from '@/api/auth'

/**
 * Returns the actions for the declared attributes section inside the agreement details.
 * @returns A function that returns an array of actions for the given attribute.
 */
export const useAgreementGetDeclaredAttributesActions = () => {
  const { t } = useTranslation('agreement', { keyPrefix: 'read.attributes' })
  const { routeKey } = useCurrentRoute()
  const { isAdmin } = AuthHooks.useJwt()
  const { partyAttributes, isAgreementEServiceMine, agreement } = useAgreementDetailsContext()

  const { mutate: declareAttribute } = AttributeMutations.useDeclarePartyAttribute()

  return (attributeId: string) => {
    // The user can declare his own attributes only in the agreement create/edit view...
    if (!agreement || routeKey !== 'SUBSCRIBE_AGREEMENT_EDIT' || !isAdmin) return []
    if (isAgreementEServiceMine) return []
    const isDeclared = isAttributeOwned('declared', attributeId, partyAttributes?.declared ?? [])
    // ... only if it is not alread declared
    if (isDeclared) return []
    // ... and only if the agreement is active, draft or suspended
    if (!['ACTIVE', 'DRAFT', 'SUSPENDED'].includes(agreement.state)) return []

    const handleDeclareAttribute = (attributeId: string) => {
      declareAttribute({
        id: attributeId,
      })
    }

    return [
      {
        label: t('declared.actions.declare'),
        action: handleDeclareAttribute,
      },
    ]
  }
}
