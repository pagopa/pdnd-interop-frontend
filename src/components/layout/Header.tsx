import React from 'react'
import { useJwt } from '@/hooks/useJwt'
import { useNavigateRouter } from '@/router'
import { assistanceLink, documentationLink, pagoPaLink } from '@/config/constants'
import { HeaderAccount, HeaderProduct } from '@pagopa/mui-italia'
import { FE_LOGIN_URL, SELFCARE_BASE_URL, SELFCARE_INTEROP_PROD_ID } from '@/config/env'
import { PartyQueries } from '@/api/party/party.hooks'
import type { PartyItem } from '@/api/party/party.api.types'
import type { PartySwitchItem } from '@pagopa/mui-italia/dist/components/PartySwitch'
import { useTranslation } from 'react-i18next'
import type { TFunction } from 'i18next'

const getPartyList = (parties: Array<PartyItem> | undefined, t: TFunction<'common'>) => {
  const partyList: Array<PartySwitchItem> = []
  if (parties) {
    partyList.push(
      ...parties.map((party) => ({
        id: party.id,
        name: party.description,
        productRole: party.userProductRoles.map((role) => t(`userProductRole.${role}`)).join(', '),
      }))
    )
  }

  return partyList
}

export const Header = () => {
  const { navigate } = useNavigateRouter()
  const { t } = useTranslation('common')
  const { jwt } = useJwt()

  const queriesOptions = {
    suspense: false,
    useErrorBoundary: false,
    retry: false,
    staleTime: Infinity,
    cacheTime: Infinity,
  }

  const { data: parties } = PartyQueries.useGetPartyList(queriesOptions)

  const partyList = getPartyList(parties, t)

  const headerAccountLoggedUser = jwt
    ? { id: jwt.uid, name: jwt.name, surname: jwt.family_name, email: '' }
    : undefined

  const goToLoginPage = () => {
    window.location.assign(FE_LOGIN_URL)
  }

  const handleSelectParty = (party: PartySwitchItem) => {
    window.location.assign(
      `${SELFCARE_BASE_URL}/token-exchange?institutionId=${party.id}&productId=${SELFCARE_INTEROP_PROD_ID}`
    )
  }

  return (
    <header>
      <HeaderAccount
        rootLink={pagoPaLink}
        loggedUser={headerAccountLoggedUser}
        onLogin={goToLoginPage}
        onLogout={() => {
          navigate('LOGOUT')
        }}
        onAssistanceClick={() => {
          window.open(assistanceLink)
        }}
        onDocumentationClick={() => {
          window.open(documentationLink, '_blank')
        }}
      />

      <HeaderProduct
        productId="interop"
        productsList={[
          {
            id: 'interop',
            title: 'InteroperabilitÃ ',
            productUrl: '',
            linkType: 'internal',
          },
        ]}
        onSelectedParty={handleSelectParty}
        partyId={jwt?.selfcareId}
        partyList={partyList}
      />
    </header>
  )
}
