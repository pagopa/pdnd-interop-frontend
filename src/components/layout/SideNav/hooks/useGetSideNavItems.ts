import React from 'react'
import { useJwt } from '@/hooks/useJwt'
import type { SideNavItemView } from '../SideNav'
import type { UserProductRole } from '@/types/party.types'
import EmailIcon from '@mui/icons-material/Email'
import type { RouteKey } from '@/router/router.types'
import uniq from 'lodash/uniq'

type Views = Record<UserProductRole, Array<SideNavItemView>>

const views: Views = {
  admin: [
    {
      routeKey: 'PROVIDE',
      id: 'provider',
      children: ['PROVIDE_ESERVICE_LIST', 'PROVIDE_AGREEMENT_LIST', 'PROVIDE_PURPOSE_LIST'],
    },
    {
      routeKey: 'SUBSCRIBE',
      id: 'subscriber',
      children: [
        'SUBSCRIBE_CATALOG_LIST',
        'SUBSCRIBE_AGREEMENT_LIST',
        'SUBSCRIBE_PURPOSE_LIST',
        'SUBSCRIBE_CLIENT_LIST',
        'SUBSCRIBE_INTEROP_M2M',
      ],
    },
  ],
  api: [
    {
      routeKey: 'PROVIDE',
      id: 'provider',
      children: ['PROVIDE_ESERVICE_LIST'],
    },
    {
      routeKey: 'SUBSCRIBE',
      id: 'subscriber',
      children: ['SUBSCRIBE_CATALOG_LIST'],
    },
  ],
  security: [
    {
      routeKey: 'SUBSCRIBE',
      id: 'subscriber',
      children: ['SUBSCRIBE_CATALOG_LIST', 'SUBSCRIBE_CLIENT_LIST', 'SUBSCRIBE_INTEROP_M2M'],
    },
  ],
}

export function useGetSideNavItems() {
  const { currentRoles } = useJwt()

  return React.useMemo(() => {
    const availableSideNavItems: Array<SideNavItemView> = []

    // For each user roles
    currentRoles.forEach((userRole) => {
      // ... take all the sidenav items for that role...
      const roleSideNavItems = views[userRole]

      roleSideNavItems.forEach(({ id: roleSideNavItemId, children }, index) => {
        const sideNavBarItemIndex = availableSideNavItems.findIndex(
          ({ id }) => id === roleSideNavItemId
        )

        // ... if the item is already on the availableSideNavItems array, just merge the children
        if (sideNavBarItemIndex > -1) {
          availableSideNavItems[sideNavBarItemIndex].children?.push(...(children ?? []))
        }

        // ... if not add the side nav item inside the availableSideNavItems array result.
        if (sideNavBarItemIndex === -1) {
          availableSideNavItems.push(roleSideNavItems[index])
        }
      })
    })

    availableSideNavItems.push({ routeKey: 'NOTIFICATION', StartIcon: EmailIcon })

    if (currentRoles.includes('admin')) {
      availableSideNavItems.push({ routeKey: 'PARTY_REGISTRY' as RouteKey })
    }

    // Remove duplicated children, if there's any.
    availableSideNavItems.forEach((_, index) => {
      if (availableSideNavItems[index]?.children) {
        availableSideNavItems[index].children = uniq(availableSideNavItems[index].children)
      }
    })

    return availableSideNavItems
  }, [currentRoles])
}
