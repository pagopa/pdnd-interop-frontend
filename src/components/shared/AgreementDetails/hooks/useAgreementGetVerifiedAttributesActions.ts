import { isAttributeOwned, isAttributeRevoked } from '@/utils/attribute.utils'
import { useAgreementDetailsContext } from '../AgreementDetailsContext'
import { useTranslation } from 'react-i18next'
import { useCurrentRoute } from '@/router'
import { useJwt } from '@/hooks/useJwt'
import { AttributeMutations } from '@/api/attribute'
import type { AttributeContainer } from '@/components/layout/containers'

/**
 * Returns the actions for the verified attributes section inside the agreement details.
 * @returns A function that returns an array of actions for the given attribute.
 */
export const useAgreementGetVerifiedAttributesActions = () => {
  const { t } = useTranslation('agreement', { keyPrefix: 'read.attributes' })
  const { mode } = useCurrentRoute()
  const { isAdmin } = useJwt()
  const { partyAttributes, isAgreementEServiceMine, agreement } = useAgreementDetailsContext()

  const { mutate: verifyAttribute } = AttributeMutations.useVerifyPartyAttribute()
  const { mutate: revokeAttibute } = AttributeMutations.useRevokeVerifiedPartyAttribute()

  const ownedVerifiedAttributes = partyAttributes?.verified ?? []

  return (attributeId: string) => {
    // The user can certify verified attributes in this view only if it is a provider...
    if (!agreement || mode === 'consumer' || !isAdmin) return []
    // ... only if the e-service does not belong to itself
    if (isAgreementEServiceMine) return []
    // ... and only if the agreement is active, pending or suspended
    if (!['ACTIVE', 'PENDING', 'SUSPENDED'].includes(agreement.state)) return []

    const attribute = ownedVerifiedAttributes.find((a) => a.id === attributeId)

    const isOwned = isAttributeOwned('verified', attributeId, ownedVerifiedAttributes)
    const isOwnedButRevoked = attribute && isAttributeRevoked('verified', attribute)

    const handleVerifyAttribute = (attributeId: string) => {
      verifyAttribute({
        partyId: agreement.consumer.id,
        id: attributeId,
      })
    }

    const handleRevokeAttribute = (attributeId: string) => {
      revokeAttibute({
        partyId: agreement.consumer.id,
        attributeId,
      })
    }

    const attributeActions: React.ComponentProps<typeof AttributeContainer>['actions'] = [
      {
        label: isOwned ? t('verified.actions.update') : t('verified.actions.verify'),
        action: handleVerifyAttribute,
      },
    ]

    if (isOwned && !isOwnedButRevoked) {
      attributeActions.push({
        label: t('verified.actions.revoke'),
        action: handleRevokeAttribute,
        color: 'error',
      })
    }

    return attributeActions
  }
}
