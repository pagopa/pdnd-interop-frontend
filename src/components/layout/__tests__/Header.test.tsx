import React from 'react'
import { Header } from '../Header'
import { renderWithApplicationContext } from '@/utils/testing.utils'
import * as useJwtHook from '@/hooks/useJwt'
import { mockUseJwt } from '__mocks__/data/user.mocks'
import { vi } from 'vitest'
import { ThemeProvider } from '@mui/material'
import { theme } from '@pagopa/interop-fe-commons'

vi.spyOn(useJwtHook, 'useJwt').mockImplementation(() => mockUseJwt())

describe('Header', () => {
  it('should match snapshot', () => {
    const tree = renderWithApplicationContext(
      <ThemeProvider theme={theme}>
        <Header />
      </ThemeProvider>,
      {
        withRouterContext: true,
        withReactQueryContext: true,
      }
    )
    expect(tree.baseElement).toMatchSnapshot()
  })
})
