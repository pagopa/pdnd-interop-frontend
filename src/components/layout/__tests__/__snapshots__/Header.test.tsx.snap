// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should match snapshot 1`] = `
<body>
  <div>
    <header>
      <div
        class="css-mmt77d-MuiStack-root"
      >
        <div
          class="MuiContainer-root css-19vyd0d-MuiContainer-root"
        >
          <div
            class="css-hxrfoi-MuiStack-root"
          >
            <a
              aria-label="Vai al sito di PagoPA S.p.A."
              class="MuiButtonBase-root MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-disableElevation css-ko92na-MuiButtonBase-root-MuiButton-root"
              href="https://www.pagopa.it"
              rel="noreferrer"
              tabindex="0"
              target="_blank"
              title="Vai al sito di PagoPA S.p.A."
            >
              PagoPA S.p.A.
            </a>
            <div
              class="css-1y7pyux-MuiStack-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-disableElevation css-166eibf-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeSmall css-cstir9-MuiButton-startIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
                    data-testid="DescriptionIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
                    />
                  </svg>
                </span>
                Documentazione
              </button>
              <button
                aria-label="Documentazione"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-u78awm-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="DescriptionIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
                  />
                </svg>
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-disableElevation css-166eibf-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeSmall css-cstir9-MuiButton-startIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
                    data-testid="HelpOutlineRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-1-4h2v2h-2zm1.61-9.96c-2.06-.3-3.88.97-4.43 2.79-.18.58.26 1.17.87 1.17h.2c.41 0 .74-.29.88-.67.32-.89 1.27-1.5 2.3-1.28.95.2 1.65 1.13 1.57 2.1-.1 1.34-1.62 1.63-2.45 2.88 0 .01-.01.01-.01.02-.01.02-.02.03-.03.05-.09.15-.18.32-.25.5-.01.03-.03.05-.04.08-.01.02-.01.04-.02.07-.12.34-.2.75-.2 1.25h2c0-.42.11-.77.28-1.07.02-.03.03-.06.05-.09.08-.14.18-.27.28-.39.01-.01.02-.03.03-.04.1-.12.21-.23.33-.34.96-.91 2.26-1.65 1.99-3.56-.24-1.74-1.61-3.21-3.35-3.47z"
                    />
                  </svg>
                </span>
                Assistenza
              </button>
              <button
                aria-label="Assistenza"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-u78awm-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="HelpOutlineRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-1-4h2v2h-2zm1.61-9.96c-2.06-.3-3.88.97-4.43 2.79-.18.58.26 1.17.87 1.17h.2c.41 0 .74-.29.88-.67.32-.89 1.27-1.5 2.3-1.28.95.2 1.65 1.13 1.57 2.1-.1 1.34-1.62 1.63-2.45 2.88 0 .01-.01.01-.01.02-.01.02-.02.03-.03.05-.09.15-.18.32-.25.5-.01.03-.03.05-.04.08-.01.02-.01.04-.02.07-.12.34-.2.75-.2 1.25h2c0-.42.11-.77.28-1.07.02-.03.03-.06.05-.09.08-.14.18-.27.28-.39.01-.01.02-.03.03-.04.1-.12.21-.23.33-.34.96-.91 2.26-1.65 1.99-3.56-.24-1.74-1.61-3.21-3.35-3.47z"
                  />
                </svg>
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-disableElevation css-1p03n5a-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                title="Esci"
                type="button"
              >
                Esci
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-sol2dz"
      >
        <div
          class="MuiContainer-root css-19vyd0d-MuiContainer-root"
        >
          <div
            class="css-176y9rv-MuiStack-root"
          >
            <div
              class="css-1yo1jo6-MuiStack-root"
            >
              <div
                class=" css-1wreeyw"
                role="button"
                tabindex="0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-shh5cd-MuiTypography-root"
                >
                  Interoperabilit√†
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
                  data-testid="ArrowDropDownRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m8.71 11.71 2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiBox-root css-jta2rl"
            />
          </div>
        </div>
      </div>
    </header>
  </div>
</body>
`;
